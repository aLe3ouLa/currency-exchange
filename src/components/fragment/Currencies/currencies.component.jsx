import React from "react";

import { Select } from "./currencies.styles";

const availableCurrencies = ["GBP", "EUR", "USD", "JPY"];

const Currencies = ({ name, value, onChange, isDisabled }) => (
  <Select
    data-testid="select"
    id={name}
    name={name}
    value={value}
    onChange={onChange}
    disabled={isDisabled}
  >
    {availableCurrencies.map((currency) => {
      return <option key={currency} value={currency} label={currency} />;
    })}
  </Select>
);

export default Currencies;
