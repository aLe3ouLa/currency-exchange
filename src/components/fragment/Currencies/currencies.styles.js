import styled from "styled-components";

export const Select = styled.select`
  width: 80px;
  padding: ${({ theme }) => theme.spacing.xxs};
  font-size: 1.6rem;
  border: 0;
  height: 34px;
  background: url(http://cdn1.iconfinder.com/data/icons/cc_mono_icon_set/blacks/16x16/br_down.png)
    no-repeat right transparent;
  appearance: none;
  background-position-x: 50px;
`;
