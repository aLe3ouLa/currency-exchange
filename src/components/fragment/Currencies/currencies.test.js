import React from "react";
import Currencies from "./currencies.component";
import { cleanup, render } from "@testing-library/react";

import { ThemeProvider } from "styled-components";
import { theme, Defaults } from "../../../styles";

describe("Change Currency tests", () => {
  afterEach(cleanup);

  const props = {
    name: "select",
    value: "EUR",
    onChange: jest.fn(),
    isDisabled: false,
  };

  it("has a valid snapshot ", () => {
    const { asFragment } = render(
      <ThemeProvider theme={theme}>
        <React.Fragment>
          <Defaults />
          <Currencies {...props} />
        </React.Fragment>
      </ThemeProvider>
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
