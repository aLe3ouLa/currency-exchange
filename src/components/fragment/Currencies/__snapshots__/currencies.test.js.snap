// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Change Currency tests has a valid snapshot  1`] = `
<DocumentFragment>
  <select
    class="sc-AxjAm kNBgxb"
    data-testid="select"
    id="select"
    name="select"
  >
    <option
      label="GBP"
      value="GBP"
    />
    <option
      label="EUR"
      value="EUR"
    />
    <option
      label="USD"
      value="USD"
    />
    <option
      label="JPY"
      value="JPY"
    />
  </select>
</DocumentFragment>
`;
